sdk/tests/deqp/modules/shared/glsShaderExecUtil.js

```diff
@@ -446,7 +446,11 @@ goog.scope(function() {
             else if (gluShaderUtil.isDataTypeMatrix(basicType)) {
                 /** @type {number} */ var numRows = gluShaderUtil.getDataTypeMatrixNumRows(basicType);
                 /** @type {number} */ var numCols = gluShaderUtil.getDataTypeMatrixNumColumns(basicType);
-                /** @type {number} */ var stride = numRows * numCols * 4;//sizeof(float);
+                // A matrix consists of several (column-major) vectors. A buffer is created for
+                // every vector in gluDrawUtil.draw() below. Data in every buffer will be tightly
+                // packed. So the stride should be 0. This is different from the code in native
+                // deqp, which use only one buffer for a matrix, the data is interleaved.
+                /** @type {number} */ var stride = 0;
```

Comment:
The WebGL deqp use multiple buffers to upload data to a matrix, see the line 118 ~ line 121 in gluDrawUtil.draw() in https://www.khronos.org/registry/webgl/sdk/tests/deqp/framework/opengl/gluDrawUtil.js. It uses a loop to create multiple buffers for one matrix. So data in every buffer is tightly packed. The stride should be 0. 

The code snippet in native deqp use only one buffer for a matrix, see the code https://android.googlesource.com/platform/external/deqp/+/deqp-dev/framework/opengl/gluDrawUtil.cpp#542. So the data in this buffer is interleaved, we should set appropriate stride and offset.
The call stack is:
gluDrawUtil::draw -> gluDrawUtil::drawFromVAOBuffers -> gluDrawUtil::drawFromBuffers